python/lsst/ts/ledprojector/led_controller.py

```diff
@@ -0,0 +1,495 @@
+    default_power : `float`
+        Default power after the led is started (Watts) in range 800-1200.
+    led_was_on : `bool`
+        Was the led commanded on, as of the most recently read LabJack data?
+    """
+
+    def __init__(
+        self,
+        config: types.SimpleNamespace,
+        log: logging.Logger | None = None,
+        status_callback: Callable | None = None,
+        simulate: bool = False,
+        make_connect_time_out: bool = False,
+    ) -> None:
+        super().__init__(
+            config=config, topics=config.topics, log=log, simulation_mode=simulate
+        )
+
+        if status_callback is not None and not inspect.iscoroutinefunction(
+            status_callback
+        ):
+            raise TypeError(
+                f"status_callback={status_callback} must be None or a coroutine"
+            )
+
+        self.config = config
+        self.log = (
+            log.getChild(type(self).__name__)
+            if log is not None
+            else logging.getLogger(type(self).__name__)
+        )
+        self.make_connect_time_out = make_connect_time_out
+        self.status_callback = status_callback
+
+        # TODO: better names
+        # We want a multiple key -> 1 value list for easy parsing
+        # We are creating a dictionary with 3 keys -> 1 value
+        # 1st key : Serial number of the LED
+        # 2nd key : 0-indexed list placement of the LED
+        # 3rd key : Labjack Channel Name of that LED
+        # First, make a list of tuples for the relevant info from the config
+        for topic in self.config.topics:
+            if topic["topic_name"] == "ledControllerItem":
+                led_topic = topic
+        led_names = led_topic["led_names"]
+        channel_names = led_topic["channel_names"]
+
+        self.log.info(f"Opening led_controller with led names {led_names}")
+        self.log.info(f"Opening led_controller with channel names {channel_names}")
+
+        tmp = list(zip(led_names, channel_names))
+        # Now make a list of labjackchannel objs using the previous list's info
+        # Note that we use enumerate to get the numerical index
+        a = [
+            LabjackChannel(serialNumber=data[0], channel=data[1], index=ind)
+            for ind, data in enumerate(tmp, start=0)
+        ]
+        # Now create our multi-key dictionary
+        # TODO: write unit test for this dictionary setup
+        self.channels: Dict[Union[str, int], LabjackChannel] = {}
+        for lbc in a:
+            self.channels.update({lbc.serial: lbc})
+            self.channels.update({lbc.index: lbc})
+            self.channels.update({lbc.channel: lbc})
+
+        self.log.info(f"led_controller dictionary {self.channels}")
+
+        # Set if connected to the labjack and state data seen,
+        # cleared otherwise.
+        self.status_event = asyncio.Event()
+        self.status_task = utils.make_done_future()
+
+    @classmethod
+    def get_config_schema(cls) -> dict[str, Any]:
```

Comment:
The BaseLabJackDataClient uses different yaml. For instance, in the BaseLabJackDataClient it requires such things like offset, and scale, which I don't think are needed at all since we are only writing true/false to a GPIO.

Another thing is that my yaml includes an "led_name" which is an additional field that we can use to identify a channel, such as the serial number to the LED